go/mysql/connection.go

```diff
@@ -49,12 +49,16 @@ func (this *ConnectionConfig) Equals(other *ConnectionConfig) bool {
 	return this.Key.Equals(&other.Key) || this.ImpliedKey.Equals(other.ImpliedKey)
 }
 
-func (this *ConnectionConfig) GetDBUri(databaseName string) string {
+func (this *ConnectionConfig) GetDBUri(databaseName string, includeMultibyte bool) string {
 	hostname := this.Key.Hostname
 	var ip = net.ParseIP(hostname)
 	if (ip != nil) && (ip.To4() == nil) {
 		// Wrap IPv6 literals in square brackets
 		hostname = fmt.Sprintf("[%s]", hostname)
 	}
-	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?interpolateParams=true&autocommit=true&charset=utf8mb4,utf8,latin1", this.User, this.Password, hostname, this.Key.Port, databaseName)
+	var multibyteCharset string
+	if includeMultibyte {
+		multibyteCharset = ",gbk,gb2312,big5,cp932,sjis"
+	}
+	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?interpolateParams=%t&autocommit=true&charset=utf8mb4,utf8,latin1%s", this.User, this.Password, hostname, this.Key.Port, databaseName, !includeMultibyte, multibyteCharset)
```

Comment:
I'd create a variable called `interpolateParams` that happens to be set to `!includeMultibyte`, and use `interpolateParams` in the `Sprintf` statement. The logic to [not] use `interpolateParams` may end up considering additional inputs to the charset.

Reply:
Thanks
We need manual close `interpolateParams` if column charset with multibyte encodings like BIG5, CP932, GB2312, GBK or SJIS ?
`riskCharset` , Is there any risk for this fix ?

Reply:
There's no risk for disabling `interpolateParams`, it is a matter of performance.